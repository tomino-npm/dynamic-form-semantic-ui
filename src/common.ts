import { FormElement, DataSet } from '@tomino/dynamic-form';
import { initCss } from '@tomino/toolbelt';

export type SignatureType = {
  comment: string;
  signature: string;
  verifiedState: 'Pending' | 'Verified' | 'Rejected';
  rejected?: boolean;
  uid: string;
  name: string;
  date: Date;
  setValue?(name: string, value: any): any;
};

export type SignatureHandlers = {
  validateForm(): boolean;
  verifySignature(uid: string, signature: string): Promise<string>;
  sign(
    source: string,
    reject: boolean,
    password: string,
    reason: string,
    submit: boolean
  ): Promise<SignatureType>;
  signatureFont(): string;
};

export type FormControlProps = {
  formControl: FormElement;
  owner: DataSet;
  handlers?: { [index: string]: any };
  readOnly: boolean;
};

export const { css, renderCss } = initCss('form-');
