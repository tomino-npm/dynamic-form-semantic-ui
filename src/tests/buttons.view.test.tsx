import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { create } from './form_query_data';
import { JSONSchema, FormDefinition, FormModel } from '@tomino/dynamic-form';
import { TestComponent } from './common';

describe('Form', () => {
  const schema: JSONSchema = {
    type: 'object',
    properties: {
      approved: {
        type: 'boolean'
      },
      value: {
        type: 'string'
      }
    },
    required: ['name']
  };

  const formDefinition: FormDefinition = create.form({
    elements: [
      create.formElement({
        row: 0,
        column: 0,
        width: 4,
        control: 'Input',
        source: 'value'
      }),
      create.formElement({
        row: 0,
        column: 4,
        width: 6,
        control: 'ApproveButton',
        source: 'approved'
      }),
      create.formElement({
        row: 0,
        column: 10,
        width: 6,
        control: 'RejectButton',
        source: 'approved'
      })
    ]
  });

  const formData = {
    description: 'Some Description'
  };

  describe('Buttons', () => {
    function componentWithData() {
      const form = new FormModel(formDefinition, schema, formData);

      // just another notation
      return <TestComponent form={form} />;
    }

    it('renders correctly', () => {
      const component = renderer.create(componentWithData());
      expect(component).toMatchSnapshot();
    });

    return { componentWithData };
  });
});
