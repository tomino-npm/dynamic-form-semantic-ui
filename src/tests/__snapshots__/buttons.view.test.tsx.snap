// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders correctly 1`] = `
<div
  className="ui segment ui form"
>
  <div
    className="ui dividing header"
  >
    Standard
  </div>
  <div
    className="ui form "
  >
    <div
      className="fields"
    >
      <div
        className="four wide field"
      >
        <div
          className="ui input"
        >
          <input
            name="value"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="six wide field"
      >
        <button
          className="ui icon primary left labeled button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="check icon"
            onClick={[Function]}
          />
          Approve
        </button>
      </div>
      <div
        className="six wide field"
      >
        <button
          className="ui red icon left labeled button"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="ban icon"
            onClick={[Function]}
          />
          Reject
        </button>
      </div>
    </div>
  </div>
  <div
    className="ui dividing header"
  >
    Readonly
  </div>
  <div
    className="ui form form-hjcaghiia"
  >
    <div
      className="fields"
    >
      <div
        className="four wide field"
      >
        <div
          className="ui disabled input"
        >
          <input
            disabled={true}
            name="value"
            onChange={[Function]}
            tabIndex={-1}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
